### PR TITLE
chore: enable eslint for geo interactions and predictions packages

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

Details see each commit of this PR.

#### Description of how you validated changes

* Unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts [hunk 1]
```diff
@@ -815,13 +822,6 @@ export class AmazonLocationServiceProvider implements GeoProvider {
...
 
-		let response: BatchDeleteGeofenceCommandOutput;
-		try {
-			response = await client.send(command);
-		} catch (error) {
-			throw error;
-		}
-
-		return response;
+		return client.send(command);
```
Comment (ashika112) [inline packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts:825]: how is removing the await not affecting anything?

Comment (HuiSF) [reply packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts:825]: Looking at the original code - it `await`s the async call to 1) get the result 2) or catch the error and re-throw it - this is no difference just return the original promise directly... 

The error was return with unnecessary await.

Comment (ashika112) [reply packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts:825]: Ah got it. Now i see it. Thanks 👍

**Review by ashika112 — APPROVED** [review]

**Review by cshfang — APPROVED** [review]

File: packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts [hunk 2]
```diff
@@ -443,29 +440,30 @@ export class AmazonLocationServiceProvider implements GeoProvider {
...
 						});
 					});
+
 					return;
 				}
 
 				// Push all successes to results
 				response.Successes?.forEach(success => {
-					const { GeofenceId, CreateTime, UpdateTime } = success;
+					const { GeofenceId: geofenceId, CreateTime, UpdateTime } = success;
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts:449]: `GeofenceId ` is declared in the upper scope already. shadow vars.

File: packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts [hunk 3]
```diff
@@ -681,17 +693,19 @@ export class AmazonLocationServiceProvider implements GeoProvider {
 						};
 						results.errors.push(errorObject);
 					});
+
 					return;
 				}
 
-				const badGeofenceIds = response.Errors.map(
+				const targetBadGeofenceIds = response.Errors.map(
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply packages/geo/src/providers/location-service/AmazonLocationServiceProvider.ts:700]: Ditto shadow vars

File: packages/predictions/src/providers/AmazonAIConvertPredictionsProvider.ts [hunk 4]
```diff
@@ -420,7 +430,7 @@ export class AmazonAIConvertPredictionsProvider {
 		region: string;
 		languageCode: string;
 	}): Promise<WebSocket> {
-		return new Promise(async (res, rej) => {
+		return new Promise((resolve, _reject) => {
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply packages/predictions/src/providers/AmazonAIConvertPredictionsProvider.ts:433]: There is no `await` in the promise executor function body. So the `async` is unnecessary.

**Pull request merged**